rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSource.java

```diff
@@ -164,6 +193,23 @@ public String getIcon() {
             }
         }
 
+        @Override
+        public boolean equals(Object o) {
+            if (this == o)
+                return true;
+            if (o == null || getClass() != o.getClass())
+                return false;
+
+            Host host = (Host) o;
+
+            return hostName != null ? hostName.equals(host.hostName) : host.hostName == null;
```

Comment:
According to `generateId`, we should also compare the node source name for these hosts to determine if they are equal (i.e. two hosts can have the same name and be part of different node sources, right).